### DISCUSSION TITLE
Feature Request: CrashlyticsLogWriter should allow setting the FirebaseCrashlytics instance

### DISCUSSION THREAD

**Discussion by svenjacobs** [touchlab/Kermit#426] [General]
Currently [CrashlyticsLogWriter](https://github.com/touchlab/Kermit/blob/96281ab581c0ff0d85ebc375b3bace675c7c5696/extensions/kermit-crashlytics/src/commonMain/kotlin/co/touchlab/kermit/crashlytics/CrashlyticsLogWriter.kt) uses `FirebaseCrashlytics.getInstance()` internally to obtain an instance of `FirebaseCrashlytics`. This call uses the default `FirebaseApp` instance.

However in our app we have multiple Firebase projects and therefore configure `FirebaseApp` manually in code. We need the possibility to pass in an instance of either `FirebaseApp` or `FirebaseCrashlytics` so that logs end up in the correct Firebase project.

One way could be that `CrashlyticsCallsActual` optionally takes this instance and that `CrashlyticsCalls` becomes a constructor argument of `CrashlyticsLogWriter`.

Note: This is written from the viewpoint of an Android application. iOS should probably receive this feature, too.

**Comment by KevinSchildhorn** [comment]
I believe this would require updates to [CrashKiOS](https://github.com/touchlab/CrashKiOS) which would be a fair bit of work. I would start with an issue on there first.

**Comment by kpgalligan** [comment]
This is, in fact, something that would need to happen first in the CrashKiOS layer. Moving to discussions. CrashKiOS itself needs some refactoring, as the way the Kotlin compiler links has changed since it was designed. In addition, the Crashlytics support itself was changed dramatically. All calls now happen through ObjC's dynamic invocation. How that is accomplished would certainly complicate passing in different Crashlytics instances (although maybe not much. Would need to look at it).
